### PR TITLE
Modernize repo

### PR DESCRIPTION
**Pull request by fraenky8**
This PR does the following:

- Updates the repo to use Go 1.20
  - Adds vendor folder and go.sum to the repo.
- Replaces the underlying library with rabbitmq/amqp091-go
- Replaces Travis with CircleCI.
  - Adds checks via golangci-lint.
  - Updates and cleans up the code base.
- Updates the docs.

### CODE REVIEW

**Review by sudarshan-reddy — APPROVED** [review]
LGTM

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  #
+  # Take a look at the official docs to check if there is a newer version
+  # available:
+  #
+  # https://golangci-lint.run/usage/install/#other-ci
+  # https://github.com/golangci/golangci-lint/releases
+  #
+  golangci-lint-version:
+    type: string
+    default: "v1.51.2"
```
Comment (adirizka7) [inline .circleci/config.yml]: Should we update this to the latest version? v1.52.2 is now available: https://github.com/golangci/golangci-lint/releases/tag/v1.52.2

File: README.md [hunk 2]
```diff
@@ -10,3 +10,38 @@ that provides or attempts to provide a universal consume subscribe pattern
 regardless of whether the client wants to use RabbitMQ, SQS, Kafka or even
 Redis.
 
+## Requirements
+
+ - Go 1.19+
```
Comment (adirizka7) [inline README.md]: Should we update to 1.20+?
```suggestion
 - Go 1.20+
```

**Review by adirizka7 — CHANGES_REQUESTED** [review]
left some queries

File: .circleci/config.yml (hunk 1 above)
Comment (fraenky8) [reply .circleci/config.yml]: Addressed in a068c86

**Review by adirizka7 — APPROVED** [review]

**Comment by adirizka7** [comment]
LGTM  ✅ Feel free to tag it @fraenky8  :octocat:

**Pull request merged**